modules/http-client/src/main/scala/pillars/httpclient/httpclient.scala

```diff
@@ -27,12 +42,80 @@ final case class HttpClient[F[_]: Async](client: org.http4s.client.Client[F])
     extends pillars.Module[F]:
     export client.*
 
+    def callSecure[SI, I, EO, O, R](
+        endpoint: Endpoint[SI, I, EO, O, R],
+        uri: Option[Uri],
+        clientOptions: Http4sClientOptions = Http4sClientOptions.default
+    )(securityInput: SI, input: I): F[Either[EO, O]] =
+        callRequest(endpoint, uri)(
+          Http4sClientInterpreter[F](clientOptions).toSecureRequest(endpoint, uri)(securityInput)(input)
+        )
+
+    end callSecure
+
+    def call[SI, I, EO, O, R](
+        endpoint: PublicEndpoint[I, EO, O, R],
+        uri: Option[Uri],
+        clientOptions: Http4sClientOptions = Http4sClientOptions.default
+    )(input: I): F[Either[EO, O]] =
+        callRequest(endpoint, uri)(Http4sClientInterpreter[F](clientOptions).toRequest(endpoint, uri)(input))
+    end call
+
+    private[this] def callRequest[EO, O](
+        endpoint: AnyEndpoint,
+        uri: Option[Uri]
+    )(interpret: (Request[F], Response[F] => F[DecodeResult[Either[EO, O]]])) =
+        import HttpClient.Error.*
+        val (request, parseResponse) = interpret
+        client
+            .run(request)
+            .use(parseResponse)
+            .flatMap:
+                case DecodeResult.Value(v)                   => v.pure[F]
+                case DecodeResult.Error(raw, error)          =>
+                    DecodingError(endpoint, uri, raw, error).raiseError[F, Either[EO, O]]
+                case DecodeResult.Missing                    => Missing(endpoint, uri).raiseError[F, Either[EO, O]]
+                case DecodeResult.Multiple(vs)               => Multiple(endpoint, uri, vs).raiseError[F, Either[EO, O]]
+                case DecodeResult.Mismatch(expected, actual) =>
+                    Mismatch(endpoint, uri, expected, actual).raiseError[F, Either[EO, O]]
+                case DecodeResult.InvalidValue(errors)       =>
+                    InvalidInput(endpoint, uri, errors).raiseError[F, Either[EO, O]]
+    end callRequest
+
 end HttpClient
 
 object HttpClient:
     def apply[F[_]](using p: Pillars[F]): Client[F] = p.module[HttpClient[F]](Key).client
     case object Key extends Module.Key:
         override def name: String = "http-client"
+
+    enum Error(endpoint: AnyEndpoint, uri: Option[Uri], val number: ErrorNumber, val message: Message)
+        extends PillarsError:
+        case DecodingError(endpoint: AnyEndpoint, uri: Option[Uri], raw: String, cause: Throwable) extends Error(
+              endpoint,
+              uri,
+              ErrorNumber(1001),
+              Message(s"Cannot decode output $raw. Cause is $cause".assume)
```

Comment:
In such situation, `Message.assume` can be used.

Reply:
Fixed 🙂